tests/frame/test_common.py

```diff
@@ -205,6 +205,32 @@ def test_anti_join(
     compare_dicts(result, expected)
 
 
+@pytest.mark.parametrize(
+    "df_raw",
+    [
+        df_polars,
+        df_lazy,
+        df_pandas,
+        df_mpd,
+    ],
+)
+@pytest.mark.parametrize(
+    ("join_key", "filter_expr", "expected"),
+    [
+        (["a"], (nw.col("b") > 5), {"a": [2], "b": [6], "z": [9]}),
+        (["b"], (nw.col("b") < 5), {"a": [1, 3], "b": [4, 4], "z": [7, 8]}),
+        (["a", "b"], (nw.col("b") < 5), {"a": [1, 3], "b": [4, 4], "z": [7, 8]}),
+    ],
+)
+def test_semi_join(
```

Comment:
as this is a new test, could we use the `constructor` fixture for this one, and maybe create a new `tests/frame/join_test` file that we can later migrate the join tests to?